shared/4-hydrogen-vm-render-script.js

```diff
@@ -147,12 +149,38 @@ async function mountHydrogen() {
   // page don't say `undefined`.
   urlRouter.attach();
 
+  //const storage = new Storage(null, window.indexedDB, null, false, localStorage, platform.logger);
+  const storageFactory = new StorageFactory(
+    null,
+    window.indexedDB,
+    window.FDBKeyRange,
+    window.localStorage
+  );
+  const storage = await storageFactory.create('1', platform.logger);
```

Comment:
Currently fails with:

```
RethrownError: Failed to render Hydrogen to string. In order to reproduce, feed in these arguments into `renderHydrogenToString(...)`:
    renderToString arguments: {"todo":"uncomment the real arguments"}
    at renderHydrogenToString (C:\Users\MLM\Documents\GitHub\element\matrix-public-archive\server\hydrogen-render\1-render-hydrogen-to-string.js:140:11)
    --- Original Error ---
    RethrownError: Child process failed with exit code 1 (somehow we saw 2 errors but we really always expect 1 error)
    0. Child process printed something to stderr: RethrownError: Child process printed something to stderr:
        at Socket.<anonymous> (C:\Users\MLM\Documents\GitHub\element\matrix-public-archive\server\hydrogen-render\1-render-hydrogen-to-string.js:47:26)
        --- Original Error ---
        no persisted storage, database can be evicted by browser
        
    1. IDBRequest failed on .: (name: AbortError)  StorageError: IDBRequest failed on .: (name: AbortError) 
        at FDBOpenDBRequest.<anonymous> (C:\Users\MLM\Documents\GitHub\element\hydrogen-web\target\lib-build\hydrogen.cjs.js:563:21)
        at invokeEventListeners (C:\Users\MLM\Documents\GitHub\element\matrix-public-archive\node_modules\fake-indexeddb\build\cjs\lib\FakeEventTarget.js:26:23)
        at FDBOpenDBRequest.dispatchEvent (C:\Users\MLM\Documents\GitHub\element\matrix-public-archive\node_modules\fake-indexeddb\build\cjs\lib\FakeEventTarget.js:99:7)
        at C:\Users\MLM\Documents\GitHub\element\matrix-public-archive\node_modules\fake-indexeddb\build\cjs\FDBFactory.js:272:19
        at C:\Users\MLM\Documents\GitHub\element\matrix-public-archive\node_modules\fake-indexeddb\build\cjs\FDBFactory.js:198:16
        at Immediate.<anonymous> (C:\Users\MLM\Documents\GitHub\element\matrix-public-archive\node_modules\fake-indexeddb\build\cjs\FDBFactory.js:154:9)
        at processImmediate (node:internal/timers:466:21)
        at ChildProcess.<anonymous> (C:\Users\MLM\Documents\GitHub\element\matrix-public-archive\server\hydrogen-render\1-render-hydrogen-to-string.js:112:25)
        --- Original Error ---
        Error: Multiple child errors listed above ^
            at ChildProcess.<anonymous> (C:\Users\MLM\Documents\GitHub\element\matrix-public-archive\server\hydrogen-render\1-render-hydrogen-to-string.js:109:35)
            at ChildProcess.emit (node:events:520:28)
            at maybeClose (node:internal/child_process:1092:16)
            at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)
            at Process.callbackTrampoline (node:internal/async_hooks:130:17)
```

Reply:
The underlying hidden error is. This was being caught and dropped in [`src/matrix/storage/idb/utils.ts#L82-L87`](https://github.com/vector-im/hydrogen-web/blob/bb5711db7eabda84ef6ef4eb943a4d64ac9ba4c3/src/matrix/storage/idb/utils.ts#L82-L87)

```
Failed to createObjectStore in database=hydrogen_session_1 TypeError: log.wrap is not a function
    at createStores (C:\Users\MLM\Documents\GitHub\element\hydrogen-web\target\lib-build\hydrogen.cjs.js:2865:14)
    at create (C:\Users\MLM\Documents\GitHub\element\hydrogen-web\target\lib-build\hydrogen.cjs.js:2812:52)
    at FDBOpenDBRequest.req.onupgradeneeded (C:\Users\MLM\Documents\GitHub\element\hydrogen-web\target\lib-build\hydrogen.cjs.js:546:13)
    at invokeEventListeners (C:\Users\MLM\Documents\GitHub\element\matrix-public-archive\node_modules\fake-indexeddb\build\cjs\lib\FakeEventTarget.js:55:25)
    at FDBOpenDBRequest.dispatchEvent (C:\Users\MLM\Documents\GitHub\element\matrix-public-archive\node_modules\fake-indexeddb\build\cjs\lib\FakeEventTarget.js:99:7)
    at waitForOthersClosed (C:\Users\MLM\Documents\GitHub\element\matrix-public-archive\node_modules\fake-indexeddb\build\cjs\FDBFactory.js:145:13)
    at runVersionchangeTransaction (C:\Users\MLM\Documents\GitHub\element\matrix-public-archive\node_modules\fake-indexeddb\build\cjs\FDBFactory.js:171:3)
    at openDatabase (C:\Users\MLM\Documents\GitHub\element\matrix-public-archive\node_modules\fake-indexeddb\build\cjs\FDBFactory.js:194:5)
    at Immediate.<anonymous> (C:\Users\MLM\Documents\GitHub\element\matrix-public-archive\node_modules\fake-indexeddb\build\cjs\FDBFactory.js:261:7)
    at processImmediate (node:internal/timers:466:21)
```

Reply:
It seems like we can catch it via `process.on('uncaughtException',` ✅